Sources/AblyChat/DefaultMessages.swift

```diff
@@ -0,0 +1,265 @@
+                    return
+                }
+
+                guard let timeSerial = try message.extras?.toJSON()["timeserial"] as? String else {
+                    return
+                }
+                
+                listener.listener(
+                    .init(type: .created,
+                          message: .init(timeserial: timeSerial,
+                                         clientID: message.clientId,
+                                         roomID: self.roomID,
+                                         text: text,
+                                         createdAt: message.timestamp,
+                                         metadata: .init(),
+                                         headers: .init())
+                         )
+                )
+            }
+        })
+        
+        return MessageSubscriptionWrapper { [self] in
+            await removeSubscriptionPoint(listener: listener)
+        } getPreviousMessages: { [self] in
+            try await getBeforeSubscriptionStart(listener: listener, params: .init())
+        }
+    }
+    
+    func get(options: QueryOptions) async throws -> any PaginatedResult<Message> {
+        try await chatAPI.getMessages(roomId: roomID, params: options)
+    }
+    
+    func send(params: SendMessageParams) async throws -> Message {
+        try await chatAPI.sendMessage(roomId: roomID, params: params)
+    }
+    
+    nonisolated func subscribeToDiscontinuities() -> Subscription<ARTErrorInfo> {
+        fatalError()
+    }
+
+    nonisolated func discontinuityDetected(reason: ARTErrorInfo?) {
+        print("Discontinuity detected: \(reason ?? .createUnknownError())")
+    }
+    
+    private func getBeforeSubscriptionStart(listener: MessageEventPayloadWrapper, params: QueryOptions) async throws -> any PaginatedResult<Message> {
+        guard let subscriptionPoint = listenerSubscriptionPoints[listener] else {
+            throw ARTErrorInfo.create(
+                withCode: 40000,
+                status: 400,
+                message: "cannot query history; listener has not been subscribed yet"
+            )
+        }
+
+        // Check the end time does not occur after the fromSerial time
+        let parseSerial = try? DefaultTimeserial.calculateTimeserial(from: subscriptionPoint)
+        if let end = params.end, end > parseSerial?.timestamp ?? 0 {
+            throw ARTErrorInfo.create(
+                withCode: 40000,
+                status: 400,
+                message: "cannot query history; end time is after the subscription point of the listener"
+            )
+        }
+
+        // Query messages from the subscription point to the start of the time window
+        return try await chatAPI.getMessages(roomId: roomID, params: .init(limit: 5))
+    }
+
+    private func makeChannel(roomId: String, realtime: RealtimeClient) -> AsyncThrowingStream<any RealtimeChannelProtocol, Error> {
```

Comment:
What does this do? Why is there a sequence of channels?

Reply:
fixed, not needed